titus-common-server/src/main/java/com/netflix/titus/runtime/endpoint/metadata/spring/CallMetadataAuthentication.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.runtime.endpoint.metadata.spring;
+
+import java.util.Collection;
+
+import com.netflix.titus.api.model.callmetadata.CallMetadata;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+
+/**
+ * A decorator for an original {@link Authentication} object, which adds resolved request {@link CallMetadata}.
```

Comment:
what does `an original Authentication object` mean? Are you wanting to convey that it is a standard concept from the Spring framework?

Reply:
I am referring here to the authenticated principal object that we are wrapping. `Authentication` interface is part of Spring security API.